src/topp/DatabaseSuitability.cpp

```diff
@@ -0,0 +1,367 @@
+      {
+        OPENMS_LOG_ERROR << "Could not compute decoy cut off. Re-ranking impossible. If you want to ignore this, set the 'force_no_re_rank' flag." << endl;
+        return INCOMPATIBLE_INPUT_DATA;
+      }
+    }
+
+    UInt64 count_db = 0;
+    UInt64 count_novo = 0;
+    UInt64 count_re_ranked = 0;
+    UInt64 count_interest = 0;
+
+    for (const auto& pep_id : pep_ids)
+    {
+      vector<PeptideHit> hits = pep_id.getHits();
+
+      if (hits.empty()) continue;
+
+      PeptideHit top_hit = hits[0];
+
+      if (top_hit.getMetaValue("target_decoy") == "decoy") continue;
+
+      // check if top hit is found in de novo protein
+      set<String> accessions = top_hit.extractProteinAccessionsSet();
+      bool is_novo = true;
+      for (const String& acc : accessions)
+      {
+        if (acc.find(Constants::UserParam::CONCAT_PEPTIDE) == String::npos)
+        {
+          is_novo = false;
+          break;
+        }
+      }
+
+      if (is_novo) // top hit is de novo hit
+      {
+        if (hits.size() == 1)
+        {
+          ++count_novo;
+          continue;
+        }
+
+        PeptideHit second_hit = hits[1];
+
+        if (second_hit.getMetaValue("target_decoy") == "decoy")
+        {
+          ++count_novo;
+          continue;
+        }
+
+        // check if second hit is db hit
+        set<String> second_accessions = top_hit.extractProteinAccessionsSet();
+        bool is_novo_too = true;
+        for (const String& acc2 : second_accessions)
+        {
+          if (acc2.find(Constants::UserParam::CONCAT_PEPTIDE) == String::npos)
+          {
+            is_novo_too = false;
+            break;
+          }
+        }
+
+        if (is_novo_too) // second hit is also de novo hit
+        {
+          ++count_novo;
+        }
+        else // second hit is db hit
+        {
+          ++count_interest;
+          // check for re-ranking
+          if (no_re_rank)
+          {
+            ++count_novo;
+            continue;
+          }
+
+          if (double(top_hit.getMetaValue("MS:1002252")) - double(second_hit.getMetaValue("MS:1002252")) <= cut_off)
+          {
+            ++count_db;
+            ++count_re_ranked;
+          }
+          else ++count_novo;
+        }
+      }
+      else ++count_db; // top hit is db hit
+    }
+
+    // spectra quality
+
+    UInt64 count_novo_seq = 0;
+    set<AASequence> unique_novo;
+
+    for (const auto& pep_id : novo_peps)
+    {
+      if (pep_id.getHits().empty()) continue;
+      ++count_novo_seq;
+      unique_novo.insert(pep_id.getHits()[0].getSequence());
+    }
+
+    //-------------------------------------------------------------
+    // writing output
+    //-------------------------------------------------------------
+
+    double quality = double(count_db) / (count_db + count_novo);
+    double id_rate = double(count_novo_seq) / count_ms2_lvl;
+
+    OPENMS_LOG_INFO << count_db << " top hits that were found in the database." << endl;
```

Comment:
maybe add the total number of identified MS2 spectra, e.g. `123 / 456 (30%) of all hits were found using the database.`
same for deNovo